src/edu/csus/ecs/pc2/ui/MultipleDataSetPane.java

```diff
@@ -572,12 +584,33 @@ public ProblemDataFiles loadDataFiles(Problem aProblem, ProblemDataFiles files,
             throw new RuntimeException("Mismatch: expecting the same number of  '" + dataExtension + "'  and  '" + answerExtension + "'  files in " + dataFileBaseDirectory + "\n (found "
                     + inputFileNames.length + "  '" + dataExtension + "'  files vs. " + answerFileNames.length + "  '" + answerExtension + "'  files)");
         }
-
+        
         SerializedFile[] inputFiles = Utilities.createSerializedFiles(dataFileBaseDirectory, inputFileNames, externalDataFiles);
         SerializedFile[] answertFiles = Utilities.createSerializedFiles(dataFileBaseDirectory, answerFileNames, externalDataFiles);
+
+        SerializedFile [] existingFiles = files.getJudgesDataFiles();
+        
+        if (existingFiles != null && existingFiles.length > 0){
+            /**
+             * Existing files present, concatenate files
+             */
+            
+            Object[] newFiles = Utilities.concatArrays(existingFiles, inputFiles);
```

Comment:
Why is the type of `newFiles` given as `Object []`?   `concatArrays()` is defined to return `SerializedFile []`; I can't see any reason for making `newFiles` an `Object []` (and then having to typecast it in the next statement.  I suggest defining `newFiles` as `SerializedFile []` (and removing the following typecast).